hoot-core/src/main/cpp/hoot/core/conflate/highway/HighwaySnapMerger.cpp

```diff
@@ -249,6 +251,9 @@ bool HighwaySnapMerger::_mergePair(const OsmMapPtr& map, ElementId eid1, Element
       boost::dynamic_pointer_cast<Way>(scraps1)->setPid(w1->getPid());
     if (scraps2 && scraps2->getElementType() == ElementType::Way)
       boost::dynamic_pointer_cast<Way>(scraps2)->setPid(w2->getPid());
+    // Reverse the way if w2 is one way and w1 isn't the similar direction as w2
+    if (OneWayCriterion().isSatisfied(w2) && !DirectionFinder::isSimilarDirection(map->shared_from_this(), w1, w2))
+      wMatch->reverseOrder();
```

Comment:
Overall, both changes are almost exactly the same.  The secondary way is checked if it is `oneway=yes` and then checked if they aren't heading in a similar direction.  Then a simple `reverseOrder` call corrects the issue.